src/variety/araf.js

```diff
@@ -0,0 +1,271 @@
+//
+//  araf.js
+//
+(function(pidlist, classbase){
+	if(typeof module==='object' && module.exports){module.exports = [pidlist, classbase];}
+	else{ pzpr.classmgr.makeCustom(pidlist, classbase);}
+}(
+['araf'], {
+//---------------------------------------------------------
+// マウス入力系
+MouseEvent:{
+	inputModes : {edit:['number','clear'],play:['border','subline']},
+	mouseinput_auto : function(){
+		if(this.puzzle.playmode){
+			if(this.mousestart || this.mousemove){
+				if(this.btn==='left' && this.isBorderMode()){ this.inputborder();}
+				else{ this.inputQsubLine();}
+			}
+		}
+		else if(this.puzzle.editmode){
+			if(this.mousestart){ this.inputqnum();}
+		}
+	}
+},
+
+//---------------------------------------------------------
+// キーボード入力系
+KeyEvent:{
+	enablemake : true
+},
+
+Cell:{
+	minnum: 0,
+	maxnum: 999
+},
+
+Board:{
+	hasborder : 1,
+
+	cols: 8,
+	rows: 8
+},
+
+AreaRoomGraph:{
+	enabled : true
+},
+
+//---------------------------------------------------------
+// 画像表示系
+
+Graphic:{
+	gridcolor_type : "DLIGHT",
+
+	bordercolor_func : "qans",
+
+	circleratio : [0.45, 0.40],
+
+
+	paint : function(){
+		this.drawBGCells();
+		this.drawDashedGrid();
+
+		// this.drawQuesNumbers();
+		this.drawCircledNumbers();
+
+		this.drawBorders();
+		this.drawBorderQsubs();
+
+		this.drawChassis();
+		this.drawTarget();
+	}
+},
+
+//---------------------------------------------------------
+// URLエンコード/デコード処理
+Encode:{
+	decodePzpr : function(type){
+		this.decodeNumber16();
+	},
+	encodePzpr : function(type){
+		this.encodeNumber16();
+	},
+
+	decodeKanpen : function(){
+		this.fio.decodeCellQnum_kanpen();
+	},
+	encodeKanpen : function(){
+		this.fio.encodeCellQnum_kanpen();
+	}
+},
+//---------------------------------------------------------
+FileIO:{
+	decodeData : function(){
+		this.decodeCellQnum();
+		this.decodeBorderAns();
+	},
+	encodeData : function(){
+		this.encodeCellQnum();
+		this.encodeBorderAns();
+	},
+
+	kanpenOpen : function(){
```

Comment:
I don't think the kanpen-related code should be there (unless this is intentional and kanpen supports araf)? As far as I understand, kanpen is a different puzzle format for some of the more common types.

So all of `FileIO` except for `{de,en}codeData` can go, as well as `Encode:{de,en}codeKanpen`.